### PR TITLE
API: integrate.simpson: allow `x` to be passed positionally

### LINKED ISSUE #21300: BUG: integrate.simpson: `x` param is keyword only
### Describe your issue.

Hello,

The call signature of `scipy.integrate.simpson` has a strange `*` as the second argument and specifically requires `simpson(y, x=x)` , as opposed to `simpson(y,x)` as was the case in previous scipy versions and as is presently the case for `scipy.integrate.trapezoid`. This extra `*` argument is [not documented](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.simpson.html). I suspect this is a bug? In any case, I think having the default second argument as the `x` array would make a lot more sense and would be in line with the call signature of `trapezoid` for example. Below is a minimal working example.

Thank you,
Ankit 

### Reproducing Code Example

```python
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import numpy as np
from scipy.integrate import simpson

x = np.linspace(0,1,100)
y = x**2

print("Int(y)dx = %.2f" %simpson(y,x))
```

### Error message

```shell
Traceback (most recent call last):
  File "/home/ankit/test.py", line 10, in <module>
    print("Int(y)dx = %.2f" %simpson(y,x))
TypeError: simpson() takes 1 positional argument but 2 were given
```

### SciPy/NumPy/Python version and system information

```shell
1.14.0 1.25.0 sys.version_info(major=3, minor=10, micro=12, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /opt/_internal/cpython-3.10.13/lib/python3.10/site-packages/scipy_openblas32/include
    lib directory: /opt/_internal/cpython-3.10.13/lib/python3.10/site-packages/scipy_openblas32/lib
    name: scipy-openblas
    openblas configuration: OpenBLAS 0.3.27.dev DYNAMIC_ARCH NO_AFFINITY Zen MAX_THREADS=64
    pc file directory: /project
    version: 0.3.27.dev
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /opt/_internal/cpython-3.10.13/lib/python3.10/site-packages/scipy_openblas32/include
    lib directory: /opt/_internal/cpython-3.10.13/lib/python3.10/site-packages/scipy_openblas32/lib
    name: scipy-openblas
    openblas configuration: OpenBLAS 0.3.27.dev DYNAMIC_ARCH NO_AFFINITY Zen MAX_THREADS=64
    pc file directory: /project
    version: 0.3.27.dev
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.12.0
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  pythran:
    include directory: ../../tmp/pip-build-env-vlr41q3a/overlay/lib/python3.10/site-packages/pythran
    version: 0.16.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /opt/python/cp310-cp310/bin/python
  version: '3.10'
```

### PR DESCRIPTION
**Pull request by lucascolley**
#### Reference issue
Closes gh-21300

#### What does this implement/fix?
`x` was made keyword-only since it is an optional parameter. However, forcing the user to write (something which will often look like) `simpson(y, x=x)` instead of `simpson(y, x)` seems like an overly-restrictive anti-pattern.

This PR relaxes to allow `x` to be passed positionally. While it isn't 100% obvious that this is the ideal move, there don't seem to be any strong opinions against it in the linked issue - whereas the current state of things has caused confusion for a user.

### CODE REVIEW

**Review by mdhaber — APPROVED** [review]
If there are no objections, please self-merge a bit before branch. (Does that work, in case I forget to come back?)

**Review by j-bowhay — APPROVED** [review]
Also seems reasonable to me

**Review by tylerjereddy — APPROVED** [review]
Alright, you have two core developer approvals and CI is passing so I'll merge it.

I don't see a reason not to adjust i.e., one test to use `x` positionally to enforce this but I won't block on it I guess.

**Pull request merged**
